packages/client/hmi-client/src/utils/project-card.ts

```diff
@@ -0,0 +1,218 @@
+/**
+ * This function returns a base64 image that is dynamically generated based on the values passed into it
+ * It requires p5.js
+ */
+
+import p5 from 'p5';
+
+// Because toDataURL is not part of the official documentation
+// https://github.com/processing/p5.js/issues/2326
+interface IRenderer extends p5.Renderer {
+	toDataURL(): string;
+}
+interface Ip5 extends p5 {
+	canvas: IRenderer;
+}
+
+interface IPlaceholderArgs {
+	contributors: number;
+	models: number;
+	datasets: number;
+	papers: number;
+}
+
+// Define variables
+let placeholderCanvas: p5 | null = null;
+const imageHeight = 133;
+const imageWidth = 208;
+let contributors: number;
+let models: number;
+let datasets: number;
+let papers: number;
+
+function drawContributors(p: p5) {
+	const maxContributors = 20;
+	const scaleFactor = 10;
+	let spacerX = maxContributors - (contributors / maxContributors) * scaleFactor;
+	let spacerY = maxContributors - (contributors / maxContributors) * scaleFactor;
+	const dotSize = spacerX / 2;
+
+	const randomShift = p.random(-60, 60);
+	const from = p.color(172 + randomShift / 4, 15, 180); // lighter color
+	const to = p.color(172 + randomShift, 20, 80); // darker color
+
+	for (let side = -1; side <= 1; side += 2) {
+		spacerX = maxContributors - (contributors / maxContributors) * scaleFactor;
+		spacerY = maxContributors - (contributors / maxContributors) * scaleFactor;
+
+		for (let iY = 10; iY < imageHeight; iY += spacerY) {
+			for (let iX = imageWidth / 2; iX < imageWidth && iX > -imageWidth; iX += spacerX * side) {
+				p.fill(p.lerpColor(from, to, iY / imageHeight));
+				p.circle(iX, iY, dotSize);
+
+				for (let h = dotSize; h > 2; h--) {
+					p.fill(
+						p.lerpColor(p.lerpColor(from, to, iY / imageHeight), p.color(0, 0, 100, 0.1), 0.25)
+					);
+					p.ellipse(iX - dotSize / 10, iY - dotSize / 10, h * 0.8, h * 0.8);
+				}
+			}
+
+			spacerY *= 1.1;
+			spacerX *= 1.1;
+		}
+	}
+}
+
+// Squiggly lines across the screen for each dataset
+function drawDatasets(p: p5) {
+	function squigglyLine(y, number, lineNumber) {
```

Comment:
Nitpick: might be cheaper (computationally) to use Perlin-noises or some kind of discrete-fourier-transform to generate squigglylines.